compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7167,22 +7167,32 @@ private STNode parsePanicKeyword() {
      *
      * <p>
      * <code>
-     * checking-expr := checking-keyword expression
-     * checking-action := checking-keyword action
+     * checking-expr := checking-keyword expression [on-fail-check]
+     * checking-action := checking-keyword action [on-fail-check]
      * </code>
      *
      * @param allowActions Allow actions
      * @param isRhsExpr    Is rhs expression
      * @return Check expression node
      */
     private STNode parseCheckExpression(boolean isRhsExpr, boolean allowActions, boolean isInConditionalExpr) {
+        startContext(ParserRuleContext.CHECKING_EXPRESSION);
         STNode checkingKeyword = parseCheckingKeyword();
         STNode expr =
                 parseExpression(OperatorPrecedence.EXPRESSION_ACTION, isRhsExpr, allowActions, isInConditionalExpr);
+        STToken onKeyword = peek();
+        STToken failKeyword = getNextNextToken();
+        STNode onFailCheck = STNodeFactory.createEmptyNode();
+        if (onKeyword.kind == SyntaxKind.ON_KEYWORD && failKeyword.kind == SyntaxKind.FAIL_KEYWORD) {
```

Comment:
Since we do not know whether `peek()` and `getNextNextToken()` are `onKeyword` and `failKeyword` naming it like that is misleading. Also, the usage of those variables are only once.

```suggestion
        STNode onFailCheck = STNodeFactory.createEmptyNode();
        if (peek().kind == SyntaxKind.ON_KEYWORD && getNextNextToken().kind == SyntaxKind.FAIL_KEYWORD) {
```